_studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp

```diff
@@ -327,7 +329,14 @@ namespace UMC_MPEG2_DECODER
     // Find units in new bitstream buffer and process them
     UMC::Status MPEG2Decoder::AddOneFrame(UMC::MediaData * source)
     {
-        // The main parsing loop
+        if (m_lastSlice)
+        {
+            UMC::Status sts = AddSlice(m_lastSlice.release());
+            if (sts == UMC::UMC_ERR_NOT_ENOUGH_BUFFER || sts == UMC::UMC_OK)
+                return sts;
```

Comment:
If status is OK we should exit and if error - proceed to execute current function?

Reply:
Currently AddSlice can return UMC::UMC_OK or UMC::UMC_ERR_NOT_ENOUGH_BUFFER or UMC::UMC_ERR_NOT_ENOUGH_DATA. The 2st two means we should exit from function. The 3rd one means we should continue parsing.

Reply:
okay